### PR TITLE
[MRG] add claude model support

### PR DESCRIPTION
**Pull request by leeeizhang**
closes https://github.com/MLSysOps/MLE-agent/issues/153

### CODE REVIEW

File: mle/model.py [hunk 1]
```diff
@@ -156,6 +156,91 @@ def stream(self, chat_history, **kwargs):
...
+        """
+        # claude has not system role in chat_history
+        # https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/system-prompts
+        system_prompt = ""
+        for idx, msg in enumerate(chat_history):
+            if msg["role"] == "system":
+                system_prompt = msg["content"]
+                chat_history.pop(idx)
+
+        # claude does not support mannual `response_format`, so we append it into system prompt
```
Comment (leeeizhang) [inline mle/model.py:230]: Hi, @huangyz0918. I found that Claude cannot generate content strictly in JSON format, even if we add the format requirements into the system prompt. Any thoughts or experiences with this?

```
claude repsonse:
Certainly! I'll analyze the user requirement and dataset, then provide a recommendation based on the latest research 
and best practices. Let me start by gathering some information.

{
    "task": "Image Classification",
    "model_or_algorithm": "ResNet8",
    "frameworks": ["PyTorch", "TensorFlow"],
    "reference": [
        "https://arxiv.org/abs/1512.03385",
        "https://arxiv.org/abs/1706.02677"
    ],
    "evaluation_metric": ["Accuracy", "F1-score"],
    "training_method": "Supervised Learning with Data Augmentation",
    "serving_method": "Model can be served using TensorFlow Serving or TorchServe",
    "device": "GPU (recommended for faster training)",
    "data_summary": "The MNIST dataset is a widely used benchmark dataset for handwritten digit recognition. It 
contains 60,000 training images and 10,000 test images of handwritten digits (0-9), each 28x28 pixels in grayscale.",
    "suggestion": "Based on the user requirement to train ResNet8 on the MNIST dataset, we suggest the following..."
}

Traceback (most recent call last):
  File "/root/miniconda3/envs/mle/bin/mle", line 33, in <module>
    sys.exit(load_entry_point('mle-agent', 'console_scripts', 'mle')())
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/root/workspace/MLE-agent/mle/cli.py", line 73, in start
    return baseline(os.getcwd(), model)
  File "/root/workspace/MLE-agent/mle/workflow/baseline.py", line 69, in baseline
    advisor_report = advisor.interact("[green]User Requirement:[/green] " + ml_requirement + "\n" + ask_data(dataset))
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 131, in interact
    self.report = self.suggest(requirement)
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 121, in suggest
    suggestions = json.loads(text)
  File "/root/miniconda3/envs/mle/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/root/miniconda3/envs/mle/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/root/miniconda3/envs/mle/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Comment (huangyz0918) [reply mle/model.py:230]: Please refer to this https://docs.anthropic.com/en/docs/test-and-evaluate/strengthen-guardrails/increase-consistency

**Review by huangyz0918 — COMMENTED** [review]
So the default model is still gpt-4o?

**Comment by leeeizhang** [comment]
> So the default model is still gpt-4o?

Yes. To support different LLMs, it is suggested to remove the default model arguments in `cli.py`, and let `load_model` and the model itself choose their default model name.

File: mle/model.py (hunk 1 above)
Comment (leeeizhang) [reply mle/model.py:230]: Thanks a lot! I just prefilled the assistant output to constrain its output JSON format. The workflow went well in my tests.
image
refer to: https://docs.anthropic.com/en/docs/test-and-evaluate/strengthen-guardrails/increase-consistency

**Comment by HuaizhengZhang** [comment]
image


Tested successfully.

**Review by HuaizhengZhang — APPROVED** [review]

**Pull request merged**